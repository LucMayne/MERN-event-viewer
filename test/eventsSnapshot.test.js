const { TextEncoder, TextDecoder } = require('util');
global.TextEncoder = TextEncoder;
global.TextDecoder = TextDecoder;

const request = require('supertest');
const express = require('express');
const app = express();
const router = require('../routes/events.js');

app.use(router);

// snapshot test
describe('Test the /event-homepage route', () => {
    test('respond with an error if no token is attached to the request', async () => {
        const response = await request(app)
            .get('/event-homepage');
        // expect a 200 OK
        expect(response.statusCode).toBe(200);
        // create a snapshot of the response body
        expect(response.body).toMatchSnapshot();
    });
});
