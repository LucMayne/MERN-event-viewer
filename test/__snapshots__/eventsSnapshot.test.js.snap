// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the /event-homepage route respond with an error if no token is attached to the request 1`] = `
{
  "message": "No token attached to the request",
}
`;
